pkg/apis/logging/v1/output_types.go

```diff
@@ -0,0 +1,114 @@
+package v1
+
+import (
+	"github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1/outputs"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Output defines a destination for log messages.
+type Output struct {
+	// Name used to refer to the output from a `pipeline`.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// OutputType is inlined with a required `type` and optional extra configuration.
+	OutputType `json:",inline"`
+
+	// URL to send log messages to.
+	//
+	// Must be an absolute URL, with a scheme. Valid URL schemes depend on `type`.
+	// Special schemes 'tcp', 'udp' and 'tls' are used for output types that don't
+	// define their own URL scheme.  Example:
+	//
+	//     { type: syslog, url: tls://syslog.example.com:1234 }
+	//
+	// TLS with server authentication is enabled by the URL scheme, for
+	// example 'tls' or 'https'.  See `secretRef` for TLS client authentication.
+	//
+	// +required
+	URL string `json:"url"`
+
+	// SecretRef refers to a `secret` object for secure communication.
+	//
+	// Client-authenticated TLS is enabled if the secret contains keys
+	// `tls.crt`, `tls.key` and `ca.crt`. Output types with password
+	// authentication will use keys `password` and `username`, not
+	// the exposed 'username@password' part of the `url`.
+	//
+	// For a `ClusterLogForwarder`, `secretRef.namespace` defaults to the
+	// `cluster-logging` namespace where secrets are normally stored.
+	//
+	// +optional
+	SecretRef corev1.SecretReference `json:"secretRef,omitempty"`
+
+	// Insecure must be true for intentionally insecure outputs.
+	// Has no function other than a marker to help avoid configuration mistakes.
+	//
+	// +optional
+	Insecure bool `json:"insecure,omitempty"`
+
+	// FIXME(alanconway) remove this unless it can be supported in initial release.
+	//
+	// Reconnect configures how the output handles connection failures.
+	// Auto-reconnect is enabled by default.
+	//
+	// +optional
+	// Reconnect *Reconnect `json:"reconnect,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Unreliable;Retry
+type Reliability string
+
+const (
+	// Unreliable may drop data after a reconnect (at-most-once).
+	Unreliable = "Unreliable"
```

Comment:
I thought the pattern for enums was like https://github.com/openshift/cluster-logging-operator/blob/master/pkg/apis/logging/v1alpha1/forwarding_types.go#L12-L15

Reply:
+1 that is the pattern we have seen used in k8s

Reply:
I can switch to lower-hyhenated if that's the way we have been doing things. I suspect it's an older convention, grepping around it looks like almost everything that uses the kubebuilder:validation:Enum marker uses MixedCaps. What's best for us?

The  kubebuilder book and k8s examples use MixedCaps
```
sigs.k8s.io/controller-tools/testData/pkg/apis/fun/v1alpha1/toy_types.go:	// +kubebuilder:validation:Enum=Lion,Wolf,Dragon
./kubernetes-sigs/controller-tools/pkg/crd/testdata/cronjob_types.go:// +kubebuilder:validation:Enum=Allow;Forbid;Replace
````
so does the elasticsearch operator:
```
github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1/elasticsearch_types.go:// +kubebuilder:validation:Enum=FullRedundancy;MultipleRedundancy;SingleRedundancy;ZeroRedundancy
```
Grepping my local ~/go/src I find a few lowercase_underscore (including bazai) but no lowercase-hypen. The majority of validation:Enum use MixedCaps:
```
[aconway@grommit src]$ find -name '*.go' -type f | xargs grep 'kubebuilder:validation:Enum:*=[a-z]' 2> /dev/null
./github.com/banzaicloud/logging-operator/pkg/sdk/model/output/format.go:	// +kubebuilder:validation:Enum=out_file;json;ltsv;csv;msgpack;hash;single_value
./github.com/openshift/installer/vendor/github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1/baremetalhost_types.go:	// +kubebuilder:validation:Enum=registration error;inspection error;provisioning error;power management error
./github.com/knative/eventing-contrib/github/pkg/apis/sources/v1alpha1/githubsource_types.go:	// +kubebuilder:validation:Enum=check_suite,commit_comment,create,delete,deployment,deployment_status,fork,gollum,installation,integration_installation,issue_comment,issues,label,member,membership,milestone,organization,org_block,page_build,ping,project_card,project_column,project,public,pull_request,pull_request_review,pull_request_review_comment,push,release,repository,status,team,team_add,watch

[aconway@grommit src]$ find -name '*.go' -type f | xargs grep 'kubebuilder:validation:Enum:*=[A-Z]' 2> /dev/null | wc -l
24
```

Reply:
"Lower hyphenated" is not the issue; Its defining struct values using an aliased type like, note LogSourceType: 
```
type LogSourceType string

const (
	LogForwardingKind string = "LogForwarding"

	//LogSourceTypeApp are container logs from non-infra structure containers
	LogSourceTypeApp LogSourceType = "logs-app"
```

Reply:
Aha, sorry. Yes will fix.